app/controllers/web/admin/domains.js

```diff
@@ -0,0 +1,49 @@
+const paginate = require('koa-ctx-paginate');
+
+const { Domains } = require('../../../models');
+
+async function list(ctx) {
+  const query = {};
+
+  // filter based on regex name
+  if (ctx.query.keyword) query.name = { $regex: ctx.query.keyword };
+
+  const [domains, itemCount] = await Promise.all([
+    Domains.find(query)
+      .limit(ctx.query.limit)
+      .skip(ctx.paginate.skip)
+      .sort(ctx.query.sort || '-created_at')
+      .lean()
+      .exec(),
+    Domains.countDocuments(query)
+  ]);
+
+  const pageCount = Math.ceil(itemCount / ctx.query.limit);
+
+  if (ctx.accepts('html'))
+    return ctx.render('admin/domains', {
+      domains,
+      pageCount,
+      itemCount,
+      pages: paginate.getArrayPages(ctx)(3, pageCount, ctx.query.page)
+    });
+
+  // this will assign rendered html to ctx.body
+  await ctx.render('admin/domains/_table', {
```

Comment:
This is an anti-pattern and bad practice for code readability to assume users know that calling `ctx.render` assigns to `ctx.body`.

Instead you should write it out here as `ctx.body = await ctx.render(...)`.

I think all we'd have to do is add a `return html` here to `koa-views` to support this @ https://github.com/queckezz/koa-views/blob/master/src/index.js#L72

I'm an author on `koa-views` on npm I think so once you submit a PR I can merge and release with semver major bump.

Reply:
We will also need to change this everywhere else we do this practice, as it's an anti-pattern to write in a way that's not readable to newbies.

Reply:
See more context here <https://titanism.slack.com/archives/C02FU8YN0L8/p1637614211010500>

Reply:
https://github.com/queckezz/koa-views/pull/161

waiting on this to be accepted.